pkg/sfu/forwarder.go

```diff
@@ -1462,37 +1553,90 @@ func (f *Forwarder) getTranslationParamsVideo(extPkt *buffer.ExtPacket, layer in
 			tp.shouldDrop = true
 			f.rtpMunger.PacketDropped(extPkt)
 			return tp, nil
+		} else if tp.switchingToTargetLayer {
+			// lock to target layer
+			f.logger.Infow("locking to target layer", "current", f.currentLayers, "target", f.targetLayers)
+			f.currentLayers.Spatial = f.targetLayers.Spatial
+			if !f.isTemporalSupported {
+				f.currentLayers.Temporal = f.targetLayers.Temporal
+			}
+			// TODO : we switch to target layer immediately now since we assume all frame chain is integrity
+			//   if we have frame chain check, should switch only if target chain is not broken and decodable
+			// if f.ddLayerSelector != nil {
+			// 	f.ddLayerSelector.SelectLayer(f.currentLayers)
+			// }
+			if f.currentLayers.Spatial >= f.maxLayers.Spatial || f.currentLayers.Spatial == (f.numAdvertisedLayers-1) {
+				tp.isSwitchingToMaxLayer = true
+			}
 		}
-	}
+	} else {
```

Comment:
Diff is impossible to read here, but the main thing here is that I am not disturbing the code path used by dependency descriptor. So, the code in the above section is the same as the old code conditioned inside `if f.ddLayerSelector != nil`